provisioner/roles/splunk_enterprise/tasks/main.yml

```diff
@@ -0,0 +1,172 @@
+        type: A
+        overwrite: yes
+        value: "{{ansible_host}}"
+      delegate_to: localhost
+      when:
+        - check_cert is failed
+  when:
+    - not teardown
+
+# tasks file for splunk_enterprise
+- name: "install epel"
+  yum:
+    name: 'https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm'
+    state: present
+
+- name: "install pre-reqs"
+  package:
+    name:
+      - rsync
+      - tar
+      - procps-ng
+      - wget
+      - net-tools
+      - curl
+      - sudo
+      - iputils
+      - bind-utils
+      - ansible
+      - python-requests
+      - git
+    state: present
+
+- name: ensure groups exist
+  group:
+    name: "{{ item }}"
+    state: present
+  loop:
+    - 'ansible'
+    - 'splunk'
+- name: ensure users exist
+  user:
+    name: "{{ item['name'] }}"
+    state: present
+    group: "{{ item['group'] }}"
+  loop:
+    - { name: 'ansible', group: 'ansible' }
+    - { name: 'splunk', group: 'splunk' }
+
+- name: ensure hardcoded splunk-ansible container dir exists
+  file:
+    path: /opt/splunk/
+    state: directory
+
+- name: INSTALL CERTBOT
+  yum:
+    name: certbot
+    enablerepo: "rhui-REGION-rhel-server-extras,rhui-REGION-rhel-server-optional"
+
+#If this fails check out status of certbot: https://letsencrypt.status.io/
+- name: ISSUE CERT
+  shell: "certbot certonly --standalone -d {{username}}-siem.{{ec2_name_prefix|lower}}.{{workshop_dns_zone}} --email ansible-security-automation@redhat.com --noninteractive --agree-tos"
+  register: issue_cert
+  until: issue_cert is not failed
+  retries: 5
+  ignore_errors: yes
+
+- name: APPEND LETS ENCRYPT FAILED
+  set_fact:
+    dns_information: |
+      - The Lets Encrypt certbot failed, please check https://letsencrypt.status.io/ to make sure the service is running
+  run_once: yes
+  delegate_to: localhost
+  delegate_facts: true
+  when: issue_cert is failed
+
+- name: MAKE SPLUNK PKI DIR
+  file:
+    path: /etc/pki/splunk
+    state: directory
+    mode: 0755
+
+- name: MOVE SSL KEY
+  copy:
+    remote_src: True
+    src: "/etc/letsencrypt/live/{{username}}-siem.{{ec2_name_prefix|lower}}.{{workshop_dns_zone}}/privkey.pem"
+    dest: /etc/pki/splunk/splunk.key
+  when: issue_cert is not failed
+
+- name: GRAB SPECIFIC SSL CERT
+  slurp:
+    src: "/etc/letsencrypt/live/{{username}}-siem.{{ec2_name_prefix|lower}}.{{workshop_dns_zone}}/cert.pem"
+  register: intermediate_cert
+
+- name: COMBINE SPECIFIC AND INTERMEDIATE CERT
+  template:
+    src: combined_cert.j2
+    dest: /etc/pki/splunk/splunk.cert
+
+- name: Download Splunk from S3
+  get_url:
+    url: 'https://s3.amazonaws.com/linklight.securityautomation/splunk-7.2.6-c0bf0f679ce9-Linux-x86_64.tgz'
+    dest: /tmp/splunk.tgz
+
+- name: Git clone splunk-ansible
+  git:
+    repo: 'https://github.com/splunk/splunk-ansible'
+    version: '7.2.6'
+    dest: '/tmp/splunk-ansible'
+
+- name: Copy splunk-ansible Extra Vars
```

Comment:
You are encapsulating an ansible-call here? That looks ugly, why is this necessary? Also, this might make it hard to re-use in Tower.

Reply:
@liquidat I am, I'm using [splunk-ansible](https://github.com/splunk/splunk-ansible) for the install of splunk but that particular set of playbooks is explicitly designed to be run on the local host of the splunk instance (yes, this is an anti-pattern and I hope to sync with that team to resolve that)

Reply:
Ah, good to know, thanks.